Env. variable added.

ph1l74 opened the pull request.

Commit:
Env. variable added.

b0g3r commented on browser-app/http.py:
```diff
@@ -1,8 +1,9 @@
-import re
+import re, os
```
BAD
https://www.python.org/dev/peps/pep-0008/#imports

Commit:
1. "while" cycle replaced to "for"
2. Comments added to browser-app
3. RegEx deleted.

Commit:
1. "while" cycle replaced to "for"
2. Comments added to browser-app
3. RegEx deleted.

b0g3r commented on browser-app/http.py:
```diff
@@ -11,25 +12,34 @@ def index():
-    while i < len(list):
-        result = hex(int(list[i], 16) ^ int(result, 16))
+
+    for i, element in enumerate(entered_text.split()):
```
я бы element переименовал бы в что-нибудь более специфичное

ph1l74 replied:
выбрал не самое лучшее название, но понятное -- hexbyte

b0g3r commented on browser-app/http.py:
```diff
@@ -11,25 +12,34 @@ def index():
-            result_string = result_string + " + " + str(hex(int(list[i], 16)))
-        i += 1
-    result = str(result)
+            result_string = result_string + " + 0x" + element
```
+=

ph1l74 replied:
Done

b0g3r commented on browser-app/http.py:
```diff
@@ -11,25 +12,34 @@ def index():
+    # Convert result string to hexdecimal view
+    result = "%X" % result
+    result = str(result).upper()
     result_string = result_string.upper() + " = "
```
result_string - всё ещё плозое название для переменной

ph1l74 replied:
изменено на package

b0g3r commented on browser-app/http.py:
```diff
@@ -11,25 +12,34 @@ def index():
+    # Convert result string to hexdecimal view
+    result = "%X" % result
+    result = str(result).upper()
     result_string = result_string.upper() + " = "
```
result_string - всё ещё плозое название для переменной

ph1l74 replied:
изменено на package

b0g3r commented on browser-app/http.py:
```diff
@@ -11,25 +12,34 @@ def index():
+
+    # Convert result string to hexdecimal view
+    result = "%X" % result
+    result = str(result).upper()
```
он уже str

b0g3r replied:
он уже upper
(см %X)

ph1l74 replied:
исправлено

b0g3r commented on browser-app/http.py:
```diff
@@ -11,25 +12,34 @@ def index():
+            result_string = result_string + " + 0x" + element
+
+    # Convert result string to hexdecimal view
+    result = "%X" % result
```
нестрогое требование от меня использовать new-style formatting через .format вместо %-style

ph1l74 replied:
Выбрал такой вариант: 
`result = "0x{0:X}" .format(result)`

Отказался от:
`result = "{0:#X}" .format(result)`
потому что нужно иметь вывод байта в формате 0x3В, а не 0X3B

b0g3r commented on window-app/main.py:
```diff
@@ -13,19 +12,17 @@
+    for i, element in enumerate(entered_text.split()):
+        result = int(element, 16) ^ result
     result = "%X" % result
     result = str(result)
```
ненада

ph1l74 replied:
Убрал)

b0g3r replied:
Пруф или зассал?

ph1l74 replied:
Готово

b0g3r commented on window-app/main.py:
```diff
@@ -13,19 +12,17 @@
     result = str(result)
     UI.lvlResult.setText(result)
-    UI.txtEnter.appendPlainText(" = " + result)
+    UI.txtEnter.setPlainText(entered_text + " " + result)
```
думаю, здесь больше пойдет `'{} {}'.format(entered_text, result)`

ph1l74 replied:
К оконному приложению -- позже. Сперва веб закончу.

ph1l74 replied:
Готово

Commit:
browser-app:
1. element -> hexbyte
2. +=
3. .format
4 - str
5. result_string -> package
6 + readme.md

b0g3r reviewed (approved):
В принципе - ок

Commit:
window-app:
1. -str
2. .format

b0g3r commented on window-app/main.py:
```diff
@@ -13,19 +12,16 @@
-    list = re.split(' ', text)
-    i = 0
+
+def convert_xor():
```
теперь выглядит почти как полное дублирование
надо выносить куда-нибудь в общий модуль

ph1l74 replied:
согласен

Commit:
browser-app:
1. + requirements.txt
window-app:
1. + requirements.txt
